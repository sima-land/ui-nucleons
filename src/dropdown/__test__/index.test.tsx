import { mount } from 'enzyme';
import { Dropdown } from '..';

describe('Dropdown', () => {
  it('should render without props', () => {
    const wrapper = mount(<Dropdown />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle props', () => {
    const wrapper = mount(
      <Dropdown className='hello' role='test-role'>
        Hello, world!
      </Dropdown>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
