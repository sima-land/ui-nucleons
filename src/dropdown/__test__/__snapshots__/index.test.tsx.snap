// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should handle props 1`] = `
<ForwardRef(Dropdown)
  className="hello"
  role="test-role"
>
  <div
    className="root hello shadow-z4 rounds-md__all"
    data-testid="dropdown"
    role="test-role"
  >
    <CustomScrollbar
      className="inner"
      overflow={
        {
          "x": "h",
          "y": "s",
        }
      }
    >
      <OverlayScrollbarsComponent
        className="custom-scrollbar inner"
        options={
          {
            "callbacks": {
              "onScrollStop": [Function],
            },
            "overflowBehavior": {
              "x": "h",
              "y": "s",
            },
          }
        }
      >
        <div
          className="os-host"
        >
          <div
            className="os-resize-observer-host"
          />
          <div
            className="os-padding"
          >
            <div
              className="os-viewport"
            >
              <div
                className="os-content"
              >
                Hello, world!
              </div>
            </div>
          </div>
          <div
            className="os-scrollbar os-scrollbar-horizontal "
          >
            <div
              className="os-scrollbar-track"
            >
              <div
                className="os-scrollbar-handle"
              />
            </div>
          </div>
          <div
            className="os-scrollbar os-scrollbar-vertical"
          >
            <div
              className="os-scrollbar-track"
            >
              <div
                className="os-scrollbar-handle"
              />
            </div>
          </div>
          <div
            className="os-scrollbar-corner"
          />
        </div>
      </OverlayScrollbarsComponent>
    </CustomScrollbar>
  </div>
</ForwardRef(Dropdown)>
`;

exports[`Dropdown should render without props 1`] = `
<ForwardRef(Dropdown)>
  <div
    className="root shadow-z4 rounds-md__all"
    data-testid="dropdown"
  >
    <CustomScrollbar
      className="inner"
      overflow={
        {
          "x": "h",
          "y": "s",
        }
      }
    >
      <OverlayScrollbarsComponent
        className="custom-scrollbar inner"
        options={
          {
            "callbacks": {
              "onScrollStop": [Function],
            },
            "overflowBehavior": {
              "x": "h",
              "y": "s",
            },
          }
        }
      >
        <div
          className="os-host"
        >
          <div
            className="os-resize-observer-host"
          />
          <div
            className="os-padding"
          >
            <div
              className="os-viewport"
            >
              <div
                className="os-content"
              />
            </div>
          </div>
          <div
            className="os-scrollbar os-scrollbar-horizontal "
          >
            <div
              className="os-scrollbar-track"
            >
              <div
                className="os-scrollbar-handle"
              />
            </div>
          </div>
          <div
            className="os-scrollbar os-scrollbar-vertical"
          >
            <div
              className="os-scrollbar-track"
            >
              <div
                className="os-scrollbar-handle"
              />
            </div>
          </div>
          <div
            className="os-scrollbar-corner"
          />
        </div>
      </OverlayScrollbarsComponent>
    </CustomScrollbar>
  </div>
</ForwardRef(Dropdown)>
`;
